src/sentry/snuba/metrics/extraction.py

```diff
@@ -746,34 +999,42 @@ def _filter(self, token: SearchFilter) -> RuleCondition:
         if not operator:
             raise ValueError(f"Unsupported operator {token.operator}")
 
-        value: Any = token.value.raw_value
+        # We propagate the filter in order to give as output a better error message with more context.
+        key: str = self._eval_search_key(token.key)
+        value: Any = self._eval_search_value(token.value)
         if operator == "eq" and token.value.is_wildcard():
             condition: RuleCondition = {
                 "op": "glob",
-                "name": _map_field_name(token.key.name),
+                "name": _map_field_name(key),
                 "value": [value],
             }
         else:
             # Special case: `x != ""` is the result of a `has:x` query, which
             # needs to be translated as `not(x == null)`.
             if token.operator == "!=" and value == "":
                 value = None
+
             if isinstance(value, str):
                 value = event_search.translate_escape_sequences(value)
+
             condition = {
                 "op": operator,
-                "name": _map_field_name(token.key.name),
+                "name": _map_field_name(key),
                 "value": value,
             }
 
-        if token.operator == "!=" or token.operator == "NOT IN":
+        # In case we have negation operators, we have to wrap them in the `not` condition.
+        if token.operator in ["!=", "NOT IN"]:
             condition = {"op": "not", "inner": condition}
 
         return condition
 
+    @staticmethod
+    def _eval_search_key(
+        search_key: SearchKey,
+    ) -> str:
+        return search_key.name
```

Comment:
This method can be trivially inlined. When reading code, it creates an indirection that doesn't provide much benefit.

Reply:
Yeah, I was torn if I wanted to keep it or not. It was there since the first iteration used the variable parsing. Will inline it.